internal/character/hanya/eidolon.go

```diff
@@ -0,0 +1,63 @@
+package hanya
+
+import (
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+const (
+	E1         = "hanya-e1"
+	E1Cooldown = "hanya-e1-cooldown"
+	E2         = "hanya-e2"
+)
+
+func init() {
+	modifier.Register(E1, modifier.Config{
+		Stacking: modifier.ReplaceBySource,
+		Listeners: modifier.Listeners{
+			OnTriggerDeath: E1HanyaAdv,
+		},
+	})
+
+	modifier.Register(E1Cooldown, modifier.Config{
+		Listeners: modifier.Listeners{
+			OnPhase2: E1HanyaCD,
+		},
+	})
+
+	modifier.Register(E2, modifier.Config{
+		StatusType: model.StatusType_STATUS_BUFF,
+		Stacking:   modifier.ReplaceBySource,
+		Duration:   1,
+	})
+}
+
+func (c *char) initEidolons() {
+	if c.info.Eidolon >= 2 {
+		c.engine.AddModifier(c.id, info.Modifier{
+			Name:   E2,
+			Source: c.id,
+			Stats: info.PropMap{
+				prop.SPDConvert: 0.2,
+			},
+		})
```

Comment:
idea: Change skill logic to check for e2 and add speed modifier with duration of 1 (and other necessary logic etc) instead. Most likely need to add after `state.endAttack()` is called, but should otherwise have the same semantics. Can check tomorrow to be sure.